### PR TITLE
Optional logger

### PR DESCRIPTION
**Pull request by 2roy999**
Changing the package so the uses of ms-logger is optional.
Also removed the dependency in express.

### CODE REVIEW

File: lib/authentication_router.js [hunk 1]
```diff
@@ -1,72 +1,97 @@
...
+  if (!token) {
+    logger.warn('Redirecting to Identity Provider: No auth token found.')
+    res.redirectToIdP()
+  } else {
+    try {
+      const verifiedUser = jwt.verify(token, secret)
+      req.user = verifiedUser
+
+      const expirationOptions = {
+        maxAge: TOKEN_EXPIRATION / 1000,
```
Comment (idanstark42) [inline lib/authentication_router.js]: Don't need to specify both. All browsers support `expires`, according to mdn. Use that.

File: lib/authentication_router.js [hunk 2]
```diff
@@ -1,72 +1,97 @@
...
+        expires: new Date(Date.now() + TOKEN_EXPIRATION)
+      }
+
+      res.setHeader('Set-Cookie', [
+        cookie.serialize(USERNAME_KEY, verifiedUser.username, expirationOptions),
+        cookie.serialize(TOKEN_KEY, token, expirationOptions)
+      ])
+
+      res.setHeader('Location', HOME_ROUTE)
```
Comment (idanstark42) [inline lib/authentication_router.js]: This is done twice. It can be a function.

**Review by idanstark42 — CHANGES_REQUESTED** [review]
I see what you did here. Can I ask when will that package be used:
1. Out side of an `express` app?
2. Without the `ms-logger`?

File: lib/authentication_router.js (hunk 1 above)
Comment (2roy999) [reply lib/authentication_router.js]: expires is deprecated. So I put both like it's done in other libraries

**Comment by 2roy999** [comment]
1. express is not the only option for servers in node, using middlewares. It's possible someone would want to write a small module in connect. It's a bad practice to bring a whole framework as a dependency to a library. 

2. To be honest, I made it because on  the raspi I didn't want to use the ms-logger so I wanted the ms-auth to work with that. Eventually, I didn't even used the ms-auth. If we want to leave the ms-looger as peer dependecy I would remove that part.

**Comment by idanstark42** [comment]
So yes, please do that. I understand there are different technologies that can replace express, but the package is meant to follow the module standard, and so it needs logs that match the standard, and the ms-logger provides that. Replacing it with another love that do not follow the standard defeats the purpose.

**Review by idanstark42 — APPROVED** [review]

**Pull request merged**